sysbuild/CMakeLists.txt

```diff
@@ -172,7 +172,9 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
       endif()
     endif()
 
-    set_property(TARGET mcuboot APPEND_STRING PROPERTY CONFIG "CONFIG_UPDATEABLE_IMAGE_NUMBER=${SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES}\n")
+    math(EXPR mcuboot_total_images "${SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES} + ${SB_CONFIG_MCUBOOT_ADDITIONAL_UPDATEABLE_IMAGES}")
+
+    set_property(TARGET mcuboot APPEND_STRING PROPERTY CONFIG "CONFIG_UPDATEABLE_IMAGE_NUMBER=${mcuboot_total_images}\n")
```

Comment:
perhaps cherry-pick as `fromtree`: https://github.com/zephyrproject-rtos/zephyr/commit/8774f15bd2fe1afb70d22ecf669f399c5e1d1ffe
and adjust this line and others to: `set_config_int(mcuboot CONFIG_UPDATEABLE_IMAGE_NUMBER ${mcuboot_total_images})`

Reply:
Will defer to after 2.8, do want to do a mass replace as `set_property()` is used in a lot of places and needs updating